### PR TITLE
[14.0][MIG] crm_project: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT30261

### CODE REVIEW

File: crm_project/wizard/crm_lead_convert2task.py [hunk 1]
```diff
@@ -19,23 +18,26 @@ def default_get(self, fields):
...
+        comodel_name="crm.lead", string="Lead", domain=[("type", "=", "lead")]
+    )
+    project_id = fields.Many2one(comodel_name="project.project", string="Project")
 
     def action_lead_to_project_task(self):
         self.ensure_one()
         # get the lead to transform
         lead = self.lead_id
-        partner_id = self._find_matching_partner()
+        partner_id = lead._find_matching_partner()
```
Comment (pedrobaeza) [inline crm_project/wizard/crm_lead_convert2task.py]: It should be `partner`, isn't it?

File: crm_project/wizard/crm_lead_convert2task.py [hunk 2]
```diff
@@ -8,7 +8,6 @@ class CrmLeadConvert2Task(models.TransientModel):
     """ wizard to convert a Lead into a Project task and move the Mail Thread """
 
     _name = "crm.lead.convert2task"
-    _inherit = "crm.partner.binding"
```
Comment (pedrobaeza) [inline crm_project/wizard/crm_lead_convert2task.py]: Are we losing partner creation on the fly in this case?

File: crm_project/security/ir.model.access.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_crm_lead_convert2task,access_crm_lead_convert2task,model_crm_lead_convert2task,project.group_project_user,1,1,1,1
```
Comment (pedrobaeza) [inline crm_project/security/ir.model.access.csv]: The one that should have permission is the sales users, not project ones.

File: crm_project/__manifest__.py [hunk 4]
```diff
@@ -4,13 +4,17 @@
 {
     "name": "Lead to Task",
     "summary": "Create Tasks from Leads/Opportunities",
-    "sequence": "19",
```
Comment (pedrobaeza) [inline crm_project/__manifest__.py]: Why removing this sequence? I would leave it, as it should have any meaning (like the module loading order). It's a pity to not have any comment about the why.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
It seems this wizard is only accessible from leads view, but it would be very interesting to have it also on opportunities view.

File: crm_project/wizard/crm_lead_convert2task.py (hunk 2 above)
Comment (victoralmau) [reply crm_project/wizard/crm_lead_convert2task.py]: This model has been removed in v14.
`_find_matching_partner()` function (which is now in `crm.lead`), `_partner_assignment()` function also takes care of creating the partner if necessary.

File: crm_project/__manifest__.py (hunk 4 above)
Comment (victoralmau) [reply crm_project/__manifest__.py]: OK, I keep it then even though we don't know the reason for this need.

**Comment by victoralmau** [comment]
Changes made.
Form (`crm.crm_lead_view_form`) view is the same for opportunities and leads.

**Review by pedrobaeza — APPROVED** [review]
The sequence thing comes from the original Odoo module:

https://github.com/odoo/odoo/blame/12.0/addons/crm_project/__manifest__.py#L6

But no clue why

**Review by CarlosRoca13 — APPROVED** [review]

**Pull request merged**